### PR TITLE
dbd: init at 1.50-unstable-2016-01-04

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/db/dbd/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "gitdurandal";
+    repo = "dbd";
+    rev = "8cf5350781b6753fcdd863148a5dcc6976e693ca";
+    hash = "sha256-b2yBZ2/Ab+SviKNlyZgdfiZ7GGZ1sonZnblD0i+vuFw=";
+  };
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}/local"
+    "CC=cc"
```
Comment (wegank) [inline pkgs/by-name/db/dbd/package.nix:19]:
```suggestion
    "CC=${stdenv.cc.targetPrefix}cc"
```

File: pkgs/by-name/db/dbd/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  ];
+
+  buildFlags = lib.optional stdenv.hostPlatform.isDarwin [ "darwin" ] ++ lib.optional stdenv.hostPlatform.isUnix [ "unix" ] ++ lib.optional stdenv.hostPlatform.isWindows [ "win32" ];
```
Comment (wegank) [inline pkgs/by-name/db/dbd/package.nix:22]: I guess this will evaluate to `[ [ "darwin" ] [ "unix" ] ]` on darwin?

Comment (D3vil0p3r) [reply pkgs/by-name/db/dbd/package.nix:22]: In theory, according to the [Makefile](https://github.com/gitdurandal/dbd/blob/master/Makefile), `unix` should evaluate only `Linux, NetBSD, FreeBSD, OpenBSD`. Maybe the developer just mislabeled.

Comment (D3vil0p3r) [reply pkgs/by-name/db/dbd/package.nix:22]: I will remove windows building because failing.

File: pkgs/by-name/db/dbd/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "8cf5350781b6753fcdd863148a5dcc6976e693ca";
+    hash = "sha256-b2yBZ2/Ab+SviKNlyZgdfiZ7GGZ1sonZnblD0i+vuFw=";
+  };
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}/local"
+    "CC=${stdenv.cc.targetPrefix}cc"
+  ];
+
+  buildFlags = lib.optional stdenv.hostPlatform.isDarwin [ "darwin" ] ++ lib.optional stdenv.hostPlatform.isUnix [ "unix" ];
```
Comment (wegank) [inline pkgs/by-name/db/dbd/package.nix:22]:
```suggestion
  buildFlags = lib.optionals stdenv.isDarwin [ "darwin" ] ++ lib.optionals (stdenv.hostPlatform.isUnix && !stdenv.isDarwin) [ "unix" ];
```

File: pkgs/by-name/db/dbd/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchFromGitHub {
+    owner = "gitdurandal";
+    repo = "dbd";
+    rev = "8cf5350781b6753fcdd863148a5dcc6976e693ca";
+    hash = "sha256-b2yBZ2/Ab+SviKNlyZgdfiZ7GGZ1sonZnblD0i+vuFw=";
+  };
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}/local"
```
Comment (wegank) [inline pkgs/by-name/db/dbd/package.nix:18]:
```suggestion
    "PREFIX=${placeholder "out"}"
```

**Review by wegank — APPROVED** [review]

**Comment by a-n-n-a-l-e-e** [comment]
you also have a PR for https://github.com/NixOS/nixpkgs/pull/290280 which seems to be the precursor to dbd. not sure why both are necessary, or any, there seem to be many ways to create encrypted connections between machines these days -- can even just run ssh as a non-root user.

**Comment by D3vil0p3r** [comment]
> you also have a PR for https://github.com/NixOS/nixpkgs/pull/290280 which seems to be the precursor to dbd. not sure why both are necessary, or any, there seem to be many ways to create encrypted connections between machines these days -- can even just run ssh as a non-root user.

Mmm if you are dealing with an hardened target machine with 22 port filtered/closed, dbd can be a good solution, since we can choose to listen on any available port we wish. About the PR above, we can proceed with dbd and close sbd.

**Comment by a-n-n-a-l-e-e** [comment]
ssh can listen on ports other than 22.

**Comment by D3vil0p3r** [comment]
> ssh can listen on ports other than 22.

I mean in a pentesting scenario: think you are a security consultant that is pentesting a customer company. The customer company deployed a SOC, IDS/IPS, network traffic monitor and so on. And it has also a firewall blocking standard ports like 22. As a pentester, I can try to inject a reverse shell that contacts pentester client that is listening by dbd. The traffic is encrypted so the network traffic monitor cannot inspect it.

**Comment by a-n-n-a-l-e-e** [comment]
there's also socat

File: pkgs/by-name/db/dbd/package.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    rev = "8cf5350781b6753fcdd863148a5dcc6976e693ca";
+    hash = "sha256-b2yBZ2/Ab+SviKNlyZgdfiZ7GGZ1sonZnblD0i+vuFw=";
+  };
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "CC=${stdenv.cc.targetPrefix}cc"
+  ];
+
+  buildFlags = lib.optionals stdenv.isDarwin [ "darwin" ] ++ lib.optionals (stdenv.hostPlatform.isUnix && !stdenv.isDarwin) [ "unix" ];
```
Comment (emilytrau) [inline pkgs/by-name/db/dbd/package.nix:22]: Can this be added to `makeFlags`?

File: pkgs/by-name/db/dbd/package.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+    description = "A Netcat-clone, designed to be portable and offer strong encryption";
```
Comment (emilytrau) [inline pkgs/by-name/db/dbd/package.nix:25]: Convention is to remove leading `A`'s
```suggestion
    description = "Netcat-clone, designed to be portable and offer strong encryption";
```

File: pkgs/by-name/db/dbd/package.nix (hunk 5 above)
Comment (D3vil0p3r) [reply pkgs/by-name/db/dbd/package.nix]: > Can this be added to `makeFlags`?

@emilytrau fixed.

**Comment by sarcasticadmin** [comment]
Result of `nixpkgs-review pr 288104` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dbd</li>
  </ul>
</details>

**Review by sarcasticadmin — APPROVED** [review]

**Comment by emilytrau** [comment]
Result of `nixpkgs-review pr 288104` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dbd</li>
  </ul>
</details>

**Pull request merged**